scripts/core/EXT_EventQueryKernelTimestamps.rst

```diff
@@ -0,0 +1,79 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_extension_event_query_kernel_timestamps:
+
+=========================================
+ Event Query Kernel Timestamps Extension
+=========================================
+
+API
+----
+
+* Functions
+
+
+    * ${x}EventQueryKernelTimestampsExt
+
+
+* Enumerations
+
+
+    * ${x}_event_query_kernel_timestamps_ext_version_t
+
+
+Event Query Kernel Timestamps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This extension enables the query of event timestamps to provide up to three views of the data, for a given event on a given device:
+
+1. Host timestamps, normalized with Device timestamps
+2. Device timestamps, normalized with Host timestamps
+3. Raw device timestamps
+
+Each view requires unique backing, provided by the caller.
+
+**Notes**
+
+- There is a slight performance implication to consider when requesting normalized timestamp data as the runtime must perform some calculations on behalf of the caller.
+- This extension is designed to compliment and eventually replace all usages of ${x}EventQueryTimestampsExp and ${x}EventQueryKernelTimestamp.
+
```

Comment:
do we need to add notes on behavior for when user uses device handles vs sub-device handles?

Reply:
Yes, I can add language for those conditions.

Reply:
> do we need to add notes on behavior for when user uses device handles vs sub-device handles?

@aravindksg : why kind of behavior are you thinking we need to explain?

Reply:
off the cuff statement: "The driver could return different values for pCount when queried with device vs sub-device handles based on underlying optimizations performed. Therefore user's should take care to both query pCount and the timestamps with appropriate device/sub-device handles. Driver shall only return timestamp data that is available for a given handle if pCount is set incorrectly by user"

something along those lines

Reply:
pCount is up to the number of sub-devices, right? maybe indicating that is enough, like "pcount may be up to the number of sub-devices"?

Reply:
`pCount` could be up to number of event packets that implementation needs to return. it could depend on different optimizations driver may do internally. so not necessarily `pCount == numSubdevices`

Reply:
thanks @aravindksg . You are right. we can add then:

"The value returned by pCount is implementation specific."